gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -159,16 +161,29 @@ public JobContext(Properties jobProps, Logger logger) throws Exception {
         : 1;
   }
 
-  protected FsDatasetStateStore createStateStore(Properties jobProps, Configuration conf) throws IOException {
-    String stateStoreFsUri =
-        jobProps.getProperty(ConfigurationKeys.STATE_STORE_FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI);
-    FileSystem stateStoreFs = FileSystem.get(URI.create(stateStoreFsUri), conf);
-    String stateStoreRootDir = jobProps.getProperty(ConfigurationKeys.STATE_STORE_ROOT_DIR_KEY);
-    if (jobProps.containsKey(ConfigurationKeys.STATE_STORE_ENABLED) &&
-        !Boolean.parseBoolean(jobProps.getProperty(ConfigurationKeys.STATE_STORE_ENABLED))) {
-      return new NoopDatasetStateStore(stateStoreFs, stateStoreRootDir);
+  protected DatasetStateStore createStateStore(Properties jobProps, Configuration conf) throws IOException {
+    boolean stateStoreEnabled = !jobProps.containsKey(ConfigurationKeys.STATE_STORE_ENABLED)
+        || Boolean.parseBoolean(jobProps.getProperty(ConfigurationKeys.STATE_STORE_ENABLED));
+
+    if (stateStoreEnabled &&
+        Boolean.parseBoolean(jobProps.getProperty(ConfigurationKeys.STATE_STORE_DB_ENABLED_KEY, "false"))) {
+      MysqlDataSource mySqlDs = new MysqlDataSource();
+
+      mySqlDs.setURL(jobProps.getProperty(ConfigurationKeys.STATE_STORE_DB_URL_KEY));
+      mySqlDs.setUser(jobProps.getProperty(ConfigurationKeys.STATE_STORE_DB_USER_KEY));
+      mySqlDs.setPassword(jobProps.getProperty(ConfigurationKeys.STATE_STORE_DB_PASSWORD_KEY));
```

Comment:
We may support encrypted passwords here:
`PasswordManager.getInstance(properties).readPassword(properties.getProperty(ConfigurationKeys.STATE_STORE_DB_PASSWORD_KEY));`